Framework/src/ServiceDiscovery.cxx

```diff
@@ -120,14 +120,19 @@ void ServiceDiscovery::runHealthServer(unsigned int port)
       io_service.reset();
       timer.expires_from_now(boost::posix_time::seconds(1));
       timer.async_wait([&](boost::system::error_code ec) {
-        if (!ec)
+        if (ec.failed()) {
+          threadInfoLogger << "async_accept error: " << ec.message() << ENDM;
           acceptor.cancel();
+        }
       });
       tcp::socket socket(io_service);
       acceptor.async_accept(socket, [&](boost::system::error_code ec) {
-        if (!ec)
+        if (ec.failed()) {
+          threadInfoLogger << "async_accept error: " << ec.message() << ENDM;
           timer.cancel();
+        }
       });
+      io_service.run();
```

Comment:
This need to be run once per process. Maybe it was removed from QC/DPL code at some point? Could you verify?

Reply:
It's seems reasonable to have it, do you have any stack traces? Or the only behaviour was that process stopped responding on Consul health checks?

Reply:
The only behaviour that I can see is disappearance of the service in Consul.